### PR TITLE
feat(JS rules): add CWE-94 AWS Lambda code injection

### PR DESCRIPTION
**Pull request by elsapet**
## Description
Add rule to check for code injection using node.js vm module (virtual machine)

Relates to #606 

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: pkg/commands/process/settings/rules/javascript/aws_lambda/code_injection.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          - runInThisContext
+          - compileFunction
+      - variable: RAW_INPUT
+        detection: javascript_aws_lambda_code_injection_user_input
+  - pattern: |
+      new vm.$<METHOD>($<...>$<RAW_INPUT>$<...>)
+    filters:
+      - variable: METHOD
+        values:
+          - Script
```
Comment (didroe) [inline pkg/commands/process/settings/rules/javascript/aws_lambda/code_injection.yml]: Should we get rid of the variable as there's only one?

**Review by didroe — COMMENTED** [review]
We have `javascript_express_eval_user_input` which is similar but there's no overlap. It would be good to use the same patterns (except the user input). Or maybe just have a single `lang` rule with it all in.

Also that express rule uses CWE-95 which is more specific. You could use that one also if it fits.

**Comment by elsapet** [comment]
Thanks @didroe - I've added the patterns and updated the CWE number. 
I'm on the fence about combining them into one `lang` rule as the inputs are specific to Express and AWS lambda, and so maybe they do belong in separate folders. Nonetheless, happy to combine them if you feel strongly 👍

File: pkg/commands/process/settings/rules/javascript/aws_lambda/code_injection.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          - runInNewContext
+          - runInThisContext
+          - compileFunction
+      - variable: RAW_INPUT
+        detection: javascript_aws_lambda_code_injection_user_input
+  - pattern: |
+      new vm.Script($<...>$<RAW_INPUT>$<...>)
+    filters:
+      - variable: RAW_INPUT
+        detection: javascript_aws_lambda_code_injection_user_input
```
Comment (didroe) [inline pkg/commands/process/settings/rules/javascript/aws_lambda/code_injection.yml:18]: We could add these to the express rule

**Review by didroe — APPROVED** [review]
I'm happy to leave as two separate rules. 

After writing lots of rules, I think the way we're organising the rules isn't really up to the job. We do a mix of either duplicating parts of rules (maintenance burden, risk of differences like in this case, etc.), or we have to violate the structure and have single rules covering multiple things (rules become large, not easy to understand why something is in a rule, etc). But not something to be dealt with right now

**Pull request merged**